micrometer-core/src/main/java/io/micrometer/core/instrument/binder/db/JooqExecuteListener.java

```diff
@@ -69,40 +70,40 @@ public void end(ExecuteContext ctx) {
 
     private void stopTimerIfStillRunning(ExecuteContext ctx) {
         Iterable<Tag> queryTags = queryTagsSupplier.get();
-        Timer.Sample sample = sampleByExecuteContext.get(ctx);
+        if (queryTags == null) return;
 
-        if (sample != null && queryTags != null) {
-            synchronized (sampleLock) {
-                sampleByExecuteContext = sampleByExecuteContext.minus(ctx);
-            }
+        Timer.Sample sample;
+        synchronized (sampleLock) {
+            sample = sampleByExecuteContext.remove(ctx);
+        }
+        if (sample == null) return;
 
-            String exceptionName = "none";
```

Comment:
everything below here is just indent related as the above early return removes the nesting